tests/templates/test_subroutines.py

```diff
@@ -1207,3 +1208,284 @@ def circuit():
             return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
 
         assert np.allclose(circuit(), expectation)
+
+
+class TestPermute:
+    """Tests for the Permute template from the pennylane.templates.subroutine module."""
+
+    def test_invalid_inputs_qnodes(self):
+        """Tests if errors are thrown for invalid permutations with QNodes."""
+
+        dev = qml.device("default.qubit", wires=4)
+
+        with pytest.raises(ValueError):
+
+            @qml.qnode(dev)
+            def permute_qubits():
+                qml.templates.Permute([0], wires=[0])
+                return qml.expval(qml.PauliZ(0))
```

Comment:
Very very minor, but might be good to move this out of the `raises` context, and restrict the context to solely the line that raises the exception:

```python
@qml.qnode(dev)
def permute_qubits():
    qml.templates.Permute([0], wires=[0])
    return qml.expval(qml.PauliZ(0))

with pytest.raises(ValueError):
    permute_qubits()
```

This helps with readability, and also makes sure we are catching the correct `.ValueError`.